pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -334,24 +331,8 @@ func PrepareRuleGroupStatuses(log log.Logger, manager state.AlertInstanceManager
 	// but in this API all rules belong to the same organization. Also filter by rule name if
 	// it was provided as a query param.
 	groupedRules := make(map[ngmodels.AlertRuleGroupKey][]*ngmodels.AlertRule)
-	for _, rule := range ruleList {
-		if len(ruleNamesSet) > 0 {
-			if _, exists := ruleNamesSet[rule.Title]; !exists {
-				continue
-			}
-		}
-
-		groupKey := rule.GetGroupKey()
-		ruleGroup := groupedRules[groupKey]
-		ruleGroup = append(ruleGroup, rule)
-		groupedRules[groupKey] = ruleGroup
-	}
-	// Sort the rules in each rule group by index. We do this at the end instead of
-	// after each append to avoid having to sort each group multiple times.
-	for _, groupRules := range groupedRules {
-		ngmodels.AlertRulesBy(ngmodels.AlertRulesByIndex).Sort(groupRules)
-	}
 
+	processGroupRules(groupedRules, ruleList, ruleNamesSet)
```

Comment:
This would read nicer if it returned the slice, and the function created the slice, e.g.

```suggestion
	groupedRules := getGroupedRules(ruleList, ruleNamesSet)
```